### PR TITLE
GF Signup : Fix post purchase upsells for paid media flow

### PR DESCRIPTION
**Pull request by ddc22**
Partially Fixes - Automattic/martech#1882

## Proposed Changes

- Since the redirect_to parameter was set as a query param for checkout. The inbuilt cart based upsells were not being activated.
- This change moves this url to the destination cookie so that eventually this is picked up by the upsell nudge component. 

## Testing Instructions
* Go through the `/setup/onboarding-media` flow 
* Purchase a premium plan
* Complete checkout
* Make sure that you are redirected to the business plan upsell

Also try any other upsell that you are familiar with on the `/start` onboarding. That is if you are familiar with such an upsell.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/checkout/get-thank-you-page-url/index.ts [hunk 1]
```diff
@@ -323,11 +323,11 @@ export default function getThankYouPageUrl( {
 		return newBlogReceiptUrl;
 	}
 
-	// disable upsell for tailored signup users
-	const isTailoredSignup = isTailoredSignupFlow( signupFlowName );
+	// disable upsell for given tailored signup users
+	const isUpsellIgnoredTailoredFlow = isUpsellIgnoredTailoredSignupFlow( signupFlowName );
```
Comment (ddc22) [inline client/my-sites/checkout/get-thank-you-page-url/index.ts]: An unrelated refactor so that the code is explicit that not all tailored flows are ignoring upsells.

Comment (southp) [reply client/my-sites/checkout/get-thank-you-page-url/index.ts]: I can see why this is done but let's revert it back :) 
This change actually inspires me to have a better understanding of what we mean by tailored flows. For the time being, we often refer "flows that implements on Stepper" with it interchangeably, but now I think it's wrong. We have plenty of other flows implemented on Stepper, but those called "tailored flows" are those tailored for specific audience, providing specific onboarding experience that's "tailored" the theme, the placeholder content, etc. for their needs. 

Thus, I found it confusing that we rename this function as "isUpsellIgnoredTailoredFlow", since it implies there is other tailored flows that upsell, while there is actually none.

Does that make sense to you?

**Review by southp — CHANGES_REQUESTED** [review]
It works as described, but I have left a concern regarding the renaming of the `isTailoredFlow` function. Please see the inline comment above :)

File: client/my-sites/checkout/get-thank-you-page-url/index.ts (hunk 1 above)
Comment (ddc22) [reply client/my-sites/checkout/get-thank-you-page-url/index.ts]: I actually was not sure whether there were other flows or not. I just thought this name `isTailoredSignupFlow` conveyed an incorrect message that all tailored flows are ignored. But if that is what we want to convey here, we can probably leave that 👍

**Comment by spraveenitpro** [comment]
<details>
<summary> Completion of Purchase without Patch: </summary>


</details>

<details>
<summary>Completion of Purchase with Patch:</summary>


</details>

File: packages/onboarding/src/utils/flows.ts
Comment (spraveenitpro) [inline packages/onboarding/src/utils/flows.ts:1]: `isUpsellIgnoredTailoredSignupFlow` doesn't seem like the best name for what this method is doing. Something like `isFlowEligibleForUpsell` may be more clear.

Comment (ddc22) [reply packages/onboarding/src/utils/flows.ts:1]: Yes  I will be  reverting that change actually :)

[f9abee1](https://github.com/Automattic/wp-calypso/pull/79326/commits/f9abee10c54dbf283e5a938aeeed7d575bbab822)

**Review by southp — APPROVED** [review]
Thanks for the fix. LGTM :shipit: 
I'm deploying this on @jdc91 's behalf since the evaluation day is now :)
